sale_subscription_financial_risk/models/sale_subscription_financial_risk_cron.py

```diff
@@ -0,0 +1,136 @@
+# Copyright (C) 2019 Amplex
+# Copyright (C) 2019 Open Source Integrators
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from datetime import date, timedelta
+from dateutil.relativedelta import relativedelta
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError
+
+
+class SaleSubscription(models.Model):
+    _inherit = 'sale.subscription'
+
+    # Calculates and returns a timedelta based on the customer's credit terms
+    def _get_delta_qty(self, qty, uom):
+        if uom == 'days':
+            delta_qty = timedelta(days=qty)
+        elif uom == 'weeks':
+            delta_qty = relativedelta(weeks=qty)
+        elif uom == 'months':
+            delta_qty = relativedelta(months=qty)
+        elif uom == 'years':
+            delta_qty = relativedelta(years=qty)
+        return delta_qty
+
+    # May be called by an account.invoice onchange/payment received or
+    # by the batch loop cron sale_subscription_financial_risk_suspend.xml
+    @api.model
+    def check_service_suspensions(self):
+        for subscription_id in self:
+            try:
+                # get list of open invoices, sorted by oldest open
+                open_invoices = subscription_id.env['account.invoice'].search(
```

Comment:
```suggestion
                open_invoices = self.env['account.invoice'].search(
```

Reply:
I am a little confused by this suggestion; as we are iterating through each ``subscription_id`` in ``self``, I believe it makes sense that we invoke ``subscription_id`` in the .env call.  This is the anchor we use throughout the rest of the method whenever we need to refer to the active object.

 @osi-scampbell had suggested this was the preferred way to accomplish, it seems like this would break functionality - am I missing something?

Reply:
@barnarj It doesn't break anything. Both solutions work. General convention is to use `self` to search, create, browse other objects. There is no need to call those methods from a specific subscription.

Reply:
@max3903 Thanks for the explanation!